### PR TITLE
Update BdevNvmeAttachController

### PR DESCRIPTION
**Pull request by derekbit**
Add ctrlr-loss-timeout-sec, reconnect-delay-sec and fast-io-fail-timeout-sec for I/O error resiliency.

### CODE REVIEW

File: pkg/spdk/client/basic.go [hunk 1]
```diff
@@ -432,15 +432,18 @@ func (c *Client) BdevRaidGetInfoByCategory(category spdktypes.BdevRaidCategory)
 //	"traddr": NVMe-oF target address: ip or BDF
 //
 //	"adrfam": NVMe-oF target adrfam: ipv4, ipv6, ib, fc, intra_host
-func (c *Client) BdevNvmeAttachController(name, subnqn, traddr, trsvcid string, trtype spdktypes.NvmeTransportType, adrfam spdktypes.NvmeAddressFamily) (bdevNameList []string, err error) {
+func (c *Client) BdevNvmeAttachController(name, subnqn, traddr, trsvcid string, trtype spdktypes.NvmeTransportType, adrfam spdktypes.NvmeAddressFamily, ctrlrLossTimeoutSec, reconnectDelaySec, fastIOFailTimeoutSec uint32) (bdevNameList []string, err error) {
```
Comment (innobead) [inline pkg/spdk/client/basic.go:435]: Need to update the doc comment.

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, but one comment needed to resolve.

File: pkg/spdk/client/basic.go (hunk 1 above)
Comment (derekbit) [reply pkg/spdk/client/basic.go:441]: Updated. Thank you.

**Review by innobead — APPROVED** [review]

File: pkg/spdk/client/basic.go [hunk 2]
```diff
@@ -432,15 +432,24 @@ func (c *Client) BdevRaidGetInfoByCategory(category spdktypes.BdevRaidCategory)
...
-			Trsvcid: trsvcid,
-			Adrfam:  adrfam,
+			Traddr:               traddr,
+			Trtype:               trtype,
+			Subnqn:               subnqn,
+			Trsvcid:              trsvcid,
+			Adrfam:               adrfam,
+			CtrlrLossTimeoutSec:  ctrlrLossTimeoutSec,
+			ReconnectDelaySec:    reconnectDelaySec,
+			FastIOFailTimeoutSec: reconnectDelaySec,
```
Comment (shuo-wu) [inline pkg/spdk/client/basic.go:452]: typo here

File: app/cmd/basic/bdev_nvme.go [hunk 3]
```diff
@@ -57,6 +57,21 @@ func BdevNvmeAttachControllerCmd() cli.Command {
 				Usage: "NVMe-oF target adrfam: \"ipv4\", \"ipv6\", \"ib\", \"fc\", \"intra_host\"",
 				Value: string(spdktypes.NvmeAddressFamilyIPv4),
 			},
+			cli.UintFlag{
+				Name:  "ctrlr-loss-timeout-sec",
+				Usage: "NVMe-oF controller loss timeout in seconds for error cases",
+				Value: 0,
```
Comment (shuo-wu) [inline app/cmd/basic/bdev_nvme.go:63]: Do we need to apply default non-zero values for these timeouts?

Comment (derekbit) [reply app/cmd/basic/bdev_nvme.go:63]: We can set some non-zero values.

Comment (shuo-wu) [reply app/cmd/basic/bdev_nvme.go:63]: Sure. would you add and use some default non-zero values in this repo?

Comment (derekbit) [reply app/cmd/basic/bdev_nvme.go:63]: Yes. Updated in https://github.com/longhorn/go-spdk-helper/pull/24/files.

File: pkg/spdk/client/basic.go (hunk 2 above)
Comment (derekbit) [reply pkg/spdk/client/basic.go:452]: Fixed. Thank you.

**Pull request merged**